hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/BaseHoodieTableServiceClient.java

```diff
@@ -174,11 +188,43 @@ private void inlineCompaction(HoodieTable table, Option<Map<String, String>> ext
   /**
    * Ensures compaction instant is in expected state and performs Log Compaction for the workload stored in instant-time.s
    *
-   * @param compactionInstantTime Compaction Instant Time
+   * @param logCompactionInstantTime Compaction Instant Time
    * @return Collection of Write Status
    */
-  protected HoodieWriteMetadata<O> logCompact(String compactionInstantTime, boolean shouldComplete) {
-    throw new UnsupportedOperationException("Log compaction is not supported yet.");
+  protected HoodieWriteMetadata<O> logCompact(String logCompactionInstantTime, boolean shouldComplete) {
```

Comment:
can we revisit all access specifiers and remove protected if its not required. later if we want to override in any of the engines, we can make them protected.

Reply:
this would be a breaking change in the event of someone else implementing custom logic in a write client outside of the hudi repo